bentoml/_internal/models/stores.py

```diff
@@ -85,14 +109,14 @@ def add(self, name: str, module: str, save_options: t.Dict[str, t.Any]):
             pass
 
         latest_path.symlink_to(path)
-        ctx = StoreCtx(path=path, version=version)
-
-        yield ctx
+        # save bentoml_model.yml
+        model_yaml = Path(path, BENTOML_MODEL_YAML)
+        with model_yaml.open("w", encoding="utf-8") as f:
+            yaml.safe_dump(_gen_model_yaml(**yaml_kwargs), f)
 
-        # TODO save yaml file after model_yaml is ready
+        yield StoreCtx(path=path, version=version)
```

Comment:
I think we should either save yaml after yield or do a clean up (deleting the model directory) if the yield failed?